src/containerapp/azext_containerapp/_validators.py

```diff
@@ -50,9 +50,10 @@ def validate_create(registry_identity, registry_pass, registry_user, registry_se
         raise InvalidArgumentValueError("--registry-identity: expected an ACR registry (*.azurecr.io) for --registry-server")
 
 
-def validate_runtime(runtime, enable_java_metrics):
-    if runtime and runtime.lower() == RUNTIME_GENERIC and enable_java_metrics is not None:
-        raise ValidationError("Not support enable Java metrics with --enable-java-metrics for generic runtime with --runtime generic")
+def validate_runtime(runtime, enable_java_metrics, enable_java_agent):
+    if runtime and runtime.lower() == RUNTIME_GENERIC and (enable_java_metrics is not None or enable_java_agent is not None):

```

Comment:
nit: Seems this condition became longer and harder to read, consider if we could extract some functions to make it more clear.
```
def validate_runtime(runtime, enable_java_metrics, enable_java_agent):
	def generic_with_java_metrics(runtime, enable_java_metrics):
		return runtime.lower() == RUNTIME_GENERIC and enable_java_metrics is not None
	def generic_with_java_agent(...):
		...
	if runtime == None:
		return
	if generic_with_java_metrics:
		raise ...
	if generic_with_java_agent:
		raise ...
```